### PR TITLE
Site Editor: refactor menu creation code

### PR DESCRIPTION
**Pull request by vindl**
## Description

Since site editor is not nested within Gutenberg submenu anymore, I think it makes sense to move it in a separate menu action for easier action filtering. All of the other items within the existing function except this one are subitems of the Gutenberg menu.

## How has this been tested?

There should be no functional changes. We should make sure that Site Editor and Gutenberg menu still work as expected.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: gutenberg.php [hunk 1]
```diff
@@ -110,6 +94,31 @@ function gutenberg_menu() {
...
+				'<span class="awaiting-mod">' . __( 'beta', 'gutenberg' ) . '</span>'
+			),
+			'edit_theme_options',
+			'gutenberg-edit-site',
+			'gutenberg_edit_site_page',
+			'dashicons-layout'
+		);
+	}
+}
+add_action( 'admin_menu', 'site_editor_menu', 9 );
```
Comment (Addison-Stavlo) [inline gutenberg.php:120]: Maybe 'gutenberg_site_editor_menu' to be more specific about where its coming from?

**Review by Addison-Stavlo — COMMENTED** [review]
I like this.

1.  Since this is a base level menu page, it makes more sense to add it on its own action as opposed to conflating it with another menu page it no longer is nested under.
2.  This is more akin to loading the post/page editor vs. being part of a plugin menu itself.
2.  Having these on separate actions makes them easier to target separately.  For example, if one wanted to hide the gutenberg plugin menu but still have the site editor button available.  This makes that much easier to achieve.

File: gutenberg.php (hunk 1 above)
Comment (vindl) [reply gutenberg.php]: Good call, updated in 98e635d07eaf06a7d5a89e32c8d97d209517a350.

**Review by gziolo — APPROVED** [review]
Yes, makes sense 👍

**Pull request merged**